lib/bundle/dsl.rb

```diff
@@ -63,12 +66,21 @@ def install
       failure.zero?
     end
 
+    def cask_args(args)
+      @cask_args = args
+    end
+
     def brew(name, options = {})
       @entries << Entry.new(:brew, name, options)
     end
 
-    def cask(name)
-      @entries << Entry.new(:cask, name)
+    def cask(name, options={})
+      if options[:args]
+        options = @cask_args.merge(options[:args])
+      else
+        options = @cask_args
+      end
```

Comment:
You should update `options[:args]` here not `options`. In fact, you can replace if condition to:
```ruby
options[:args] = @cask_args.merge options.fetch(:args, {})